source/mir/random/nonuniform.d

```diff
@@ -0,0 +1,246 @@
+/**
+Non-uniform distributions.
+
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Sebastian Wilzbach
+*/
+
+module mir.random.nonuniform;
+
+import std.traits : isNumeric;
+import std.stdio;
+
+/**
+Setup a _normal distribution sampler.
+
+Params:
+    T = floating-point type that should be sampled
+    UIntType = unsigned UIntType of the random generator
+
+Returns:
+    A $(LREF Ziggurat) sampler that can be called to sample from the distribution.
+*/
+auto normal(T, UIntType = uint)()
+{
+    import mir.internal.math : exp, log, sqrt;
+
+    auto pdf    = (T x) => cast(T) exp(T(-0.5) * x * x);
+    auto invPdf = (T x) => cast(T) sqrt(T(-2) * log(x));
+
+    // values from [Marsaglia00]
+    T rightEnd = 3.442619855899;
+
+    /// generate x for the last block (aka tail-block)
+    enum fallback = q{
+        T fallback(RNG)(bool isPositive, ref RNG gen) const
+        {
+            import std.random : uniform;
+            import std.math : log;
+            T x, y, u;
+            do
+            {
+                u = uniform!("[]", T, T)(0, 1, gen);
```

Comment:
Marsaglia & Tang talk about `UNI` (in their paper) being a generator of "uniform (0, 1) variates".  That would suggest over the open rather than closed interval, i.e. `uniform!"()"` rather than `uniform!"[]"`.  Does that make a difference to your distribution results?